llama-index-integrations/llms/llama-index-llms-premai/llama_index/llms/premai/base.py

```diff
@@ -177,21 +177,57 @@ def _get_all_kwargs(self, **kwargs) -> Dict[str, Any]:
                 all_kwargs.pop(key, None)
         return all_kwargs
 
-    @llm_chat_callback()
-    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
-        all_kwargs = self._get_all_kwargs(**{**self.additional_kwargs, **kwargs})
+    def _prepare_messages_before_chat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ):
         chat_messages = []
+        all_kwargs = self._get_all_kwargs(**{**self.additional_kwargs, **kwargs})
 
         for message in messages:
             if "system_prompt" in all_kwargs and message.role.value == "system":
                 continue
-
             elif "system_prompt" not in all_kwargs and message.role.value == "system":
                 all_kwargs["system_prompt"] = message.content
-            else:
+            elif message.role.value == "assistant":
                 chat_messages.append(
                     {"role": message.role.value, "content": message.content}
                 )
+            elif message.role.value == "user":
```

Comment:
You could use `message.role == MessageRole.USER`

Reply:
Done